src/core/thread.d

```diff
@@ -2309,8 +2308,23 @@ private __gshared uint suspendDepth = 0;
  * Throws:
  *  ThreadError if the suspend operation fails for a running thread.
  */
-private void suspend( Thread t ) nothrow
+private bool suspend( Thread t ) nothrow
 {
+    Duration waittime = dur!"usecs"(10);
+ Lagain:
+    // check isRunning in case thread was terminated abnormally
+    if (!t.isRunning)
+    {
+        Thread.remove(t);
+        return false;
+    }
+    else if (atomicLoad(t.m_isInCriticalRegion))
```

Comment:
> It could have been changed after the check here and before the actual suspension.

It shouldn't and that was a bug introduced by the other pull.
See #1164 for a fix.